components/reminder/ReminderForm.js

```diff
@@ -17,19 +18,33 @@ export default function ReminderForm({ userId, reminderId }) {
 
   const quickActions = ["Water", "Fertilise", "Repot"];
   const [title, setTitle] = useState(reminder?.title ?? "");
-  const [plantId, setPlantId] = useState(reminder?.plantId ?? "");
+  const [plantId, setPlantId] = useState(
+    reminder
+      ? typeof reminder.plantId === "object"
+        ? reminder.plantId._id
+        : reminder.plantId
+      : ""
```

Comment:
here the plantId._id should be enough after our fixing
```suggestion
    reminder.plantId?._id
```